cadquery/occ_impl/shapes.py

```diff
@@ -2379,6 +2381,75 @@ def chamfer2D(self, d: float, vertices: Iterable[Vertex]) -> "Wire":
 
         return f.chamfer2D(d, vertices).outerWire()
 
+    def fillet(
+        self, radius: float, vertices: Optional[Iterable[Vertex]] = None
+    ) -> "Wire":
+        """
+
+        Apply 2D or 3D fillet to a wire
+
+        :param wire: The input wire to fillet. Currently only open wires are supported
+        :param radius: the radius of the fillet, must be > zero
+        :param vertices: Optional list of vertices to fillet. By default all vertices are fillet.
+
+        :return: A wire with filleted corners
+        """
+        edges = list(self)
+        newEdges = []
+        currentEdge = edges[0]
+
+        # Make vertices into a set so that we can query for membership
+        # in O(log n).
+        if vertices is not None:
```

Comment:
you should simply use vertices as-is an convert them to set

Reply:
Sorry. I don't understand.  `Vertex` isn't immutable, and you can't put it into python set. Which is why I'm doing the casting to an immutable tuple. Did I miss something?

Reply:
I applied my suggestions, let's see if it works

Reply:
I checked the immutability aspect of Vertex and it seems I was wrong. It's just a wrapper for `TopoDS_Vertex` which seems to be immutable. So using sets on Vertex is ok. 

However using vertices directly without casting to Vertex or tuple will fail if the user passes a list for coordinates to vertices.

I.e. the following code worked before your change, but will fail after your change:

```
        points = [
            [0.000, 0.000, 0.000],
            [-0.287, 1.183, -0.592],
            [-1.404, 4.113, -2.787],
            [-1.332, 1.522, 0.553],
            [7.062, 0.433, -0.097],
            [8.539, -0.000, -0.000],
        ]
        wire = Wire.makePolygon(points, close=False)

        # Fillet a single vertex
        wfillet = wire.fillet(radius=0.560, vertices=[points[1]])
        assert len(wfillet.Edges()) == len(points)

```

Reply:
I'd rather do a potentially redundant cast in the implementation to catch a casual misuse of the API than having the code raise a cryptic exception. But if it's fine with you, I'll yield.

Reply:
The API is specified by the type annotations, IMO nothing is cryptic here. The general vision is that you use selectors to specify fillets and chamfers.